### PR TITLE
Lunarfusion/css 107 field group core tokens

### PR DESCRIPTION
**Pull request by lunarfusion**
## Description
BREAKING CHANGE: This migrates the Radio button component to core tokens.
- Jira issue CSS-107 https://jira.corp.adobe.com/browse/CSS-107
- Adds fieldgroup label with top and side aligned layouts
- Adds nested input layout classes for horizontal and vertical inputs
- Adds required and optional variants
- Incorporates helptext dependency
- Adds Spectrum and Express themes
- Adds examples and documentation to the YML

## How and where has this been tested?
 - Chrome 103 for macOS
 - Firefox 102 for macOS
 - **How this was tested:** comparing local build of http://localhost:3000/docs/fieldgroup.html with XD redline components batch and with https://spectrum.adobe.com/page/checkbox-group/ and https://spectrum.adobe.com/page/radio-group/

## Screenshots
Vertically aligned inputs
Screen Shot 2022-07-28 at 10 30 57 AM

Horizontally aligned inputs
Screen Shot 2022-07-28 at 10 31 04 AM

Required and optional
Screen Shot 2022-07-28 at 10 31 47 AM

Side label with horizontal and vertical inputs
Screen Shot 2022-07-28 at 10 32 25 AM

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [x] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [ ] I have tested these changes in Windows High Contrast mode.

- [ ] This pull request is ready to merge.

### CODE REVIEW

**Review by bernhard-adobe — APPROVED** [review]
Looks good to me.

File: components/fieldgroup/index.css [hunk 1]
```diff
@@ -10,40 +10,78 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+ * radio, checkbox, helptext, fieldlabel
+ *
+ */
+
+/* custom properties */
 .spectrum-FieldGroup {
-  --spectrum-fieldgroup-margin: var(--spectrum-global-dimension-size-200);
-  --spectrum-fieldgroup-readonly-delineator: '\002c';
+  --spectrum-fieldgroup-margin: var(--spectrum-spacing-300);
+  --spectrum-fieldgroup-readonly-delimiter: '\002c';
```
Comment (bernhard-adobe) [inline components/fieldgroup/index.css:24]: no more delimiter?

File: components/fieldgroup/index.css [hunk 2]
```diff
@@ -10,40 +10,78 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
+  &--vertical {
+    flex-direction: column;
+  }
+
+  /* input fields aligned horizontally */
+  &--horizontal {
+    flex-direction: row;
+
+    /* space between horizontal fields */
+    .spectrum-FieldGroup-item:not(:last-child) {
```
Comment (bernhard-adobe) [inline components/fieldgroup/index.css:60]: nit, could we nest some of these definitions? 
It's also single line for a single rule so maybe, maybe not. Up to you, same below

File: components/fieldgroup/index.css [hunk 3]
```diff
@@ -10,40 +10,78 @@ OF ANY KIND, either express or implied. See the License for the specific languag
...
     .spectrum-Checkbox-box {
       display: none;
     }
 
+    /* read-only checkbox fields delimited by commas */
     &:not(:last-child) {
       .spectrum-Checkbox-label {
         &:after {
-          content: var(--spectrum-fieldgroup-readonly-delineator)
+          content: var(--spectrum-fieldgroup-readonly-delimiter)
```
Comment (bernhard-adobe) [inline components/fieldgroup/index.css:82]: ah, now I see it. lol. Subtle difference between a delineator and delimiter. 
Screen Shot 2022-08-09 at 22 04 48

File: components/fieldgroup/package.json [hunk 4]
```diff
@@ -19,18 +19,20 @@
...
+    "@spectrum-css/helptext": "^1.0.20",
+    "@spectrum-css/tokens": "^1.0.1"
   },
   "devDependencies": {
+    "@spectrum-css/component-builder-simple": "^1.0.0-beta",
     "@spectrum-css/checkbox": "^3.1.3",
-    "@spectrum-css/component-builder": "^3.2.0",
     "@spectrum-css/icon": "^3.0.23",
     "@spectrum-css/radio": "^3.0.24",
+    "@spectrum-css/helptext": "^1.0.20",
```
Comment (bernhard-adobe) [inline components/fieldgroup/package.json:30]: @pfulton we need to double-check this is the correct version after the beta-release

**Review by bernhard-adobe — APPROVED** [review]
Great work!

**Comment by bernhard-adobe** [comment]
VRT is: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/36/artifact/backstop_data/html_report/index.html
[2022-08-10 -- 1486 - field-group.zip](https://github.com/adobe/spectrum-css/files/9306279/2022-08-10.--.1486.-.field-group.zip)

**Review by pfulton — APPROVED** [review]

**Comment by pfulton** [comment]
Released: `- @spectrum-css/fieldgroup@4.0.0-beta.1`

**Comment by pfulton** [comment]
New beta release is: `@spectrum-css/fieldgroup@4.0.0-beta.2`

**Pull request merged**